avm/res/db-for-my-sql/flexible-server/tests/e2e/max/dependencies2.bicep

```diff
@@ -75,15 +76,16 @@ resource geoBackupKeyVault 'Microsoft.KeyVault/vaults@2023-02-01' = {
     }
     tenantId: tenant().tenantId
     enablePurgeProtection: true // Required for encryption to work
-    softDeleteRetentionInDays: 7
+    softDeleteRetentionInDays: 90
     enabledForTemplateDeployment: true
     enabledForDiskEncryption: true
     enabledForDeployment: true
     enableRbacAuthorization: true
+    enableSoftDelete: true
```

Comment:
Same comment as above

Reply:
Fixed.

Reply:
Hey @hundredacres, I see you reverted the values again which begs the same question as before - is that a service requirement? If so, shame on the RP 😄

Reply:
My tests started failing after making edits so I'm trying to figure out which change is required. Still working through things.

Reply:
@AlexanderSehr Ok, removed the enableSoftDelete param and my tests are working, but when passing a 7 for the retentionDays, I get an error. Still working through these.

Reply:
Hey @hundredacres,
I observed your fight from the sidelines - but I think you can call it a day now. I just found the following statement [here](https://learn.microsoft.com/en-us/azure/mysql/flexible-server/concepts-customer-managed-key#:~:text=Enable%20the%20soft%2Ddelete%20feature%20on%20the%20key%20vault%20with%20a%20retention%20period%20set%20to%2090%20days%20to%20protect%20from%20data%20loss%20should%20an%20accidental%20key%20(or%20Key%20Vault)%20deletion%20occur.):
> _Enable the [soft-delete](https://learn.microsoft.com/en-us/azure/key-vault/general/soft-delete-overview) feature on the key vault with a retention period set to 90 days to protect from data loss should an accidental key (or Key Vault) deletion occur._

This is subjectively an absolutely mental enforced requirement, but it is what it is. I can only assume this was enforced through newer API versions (as it wasn't failing before). 
If you agree, we should add the reference to the docs page to the `dependencies.bicep` file as a comment next to the property to not have somebody else try and get the same thing working. 
In any case - we'll collect A LOT of key vaults.

Reply:
@AlexanderSehr The latest commit removes the retentionDays param and lets the default value be applied, and we have passing tests! Do you still think we need a comment, or just move on?

Reply:
Hey @hundredacres,
I think it would be good to have that comment as the default retention time is 90 days 🥲 Let's set the explicit value as you did before so that it's obvious, add the reference and be done with it.

Reply:
@AlexanderSehr Let me know if the latest change works.

Reply:
Extended it ever so slightly. If you agree, then let's apply the suggestions and merge the PR 💪 Thanks @hundredacres